Content.Server/Body/Systems/RespiratorSystem.cs

```diff
@@ -222,36 +178,25 @@ private void StopSuffocation(EntityUid uid, RespiratorComponent respirator)
             _damageableSys.TryChangeDamage(uid, respirator.DamageRecovery, true);
         }
 
-        public GasMixture Clean(EntityUid uid, RespiratorComponent respirator, BloodstreamComponent bloodstream)
+        public void UpdateSaturation(EntityUid uid, float amount,
+            RespiratorComponent? respirator = null)
         {
-            var gasMixture = new GasMixture(bloodstream.Air.Volume)
-            {
-                Temperature = bloodstream.Air.Temperature
-            };
-
-            for (Gas gas = 0; gas < (Gas) Atmospherics.TotalNumberOfGases; gas++)
-            {
-                float amount;
-                var molesInBlood = bloodstream.Air.GetMoles(gas);
-
-                if (!respirator.NeedsGases.TryGetValue(gas, out var needed))
-                {
-                    amount = molesInBlood;
-                }
-                else
-                {
-                    var overflowThreshold = needed * 5f;
-
-                    amount = molesInBlood > overflowThreshold
-                        ? molesInBlood - overflowThreshold
-                        : 0;
-                }
-
-                gasMixture.AdjustMoles(gas, amount);
-                bloodstream.Air.AdjustMoles(gas, -amount);
-            }
+            if (!Resolve(uid, ref respirator, false))
+                return;
 
-            return gasMixture;
+            respirator.Saturation += amount;
+            respirator.Saturation =
+                Math.Clamp(respirator.Saturation, respirator.MinSaturation, respirator.MaxSaturation);
         }
     }
 }
+
+public class InhaleLocationEvent : EntityEventArgs
+{
+    public GasMixture? Gas;
+}
+
+public class ExhaleLocationEvent : EntityEventArgs
+{
+    public GasMixture? Gas;
+}
```

Comment:
Nothing currently subscribes to exhale events. Assuming it's not accidental, I guess it makes sense for respirators to not add the CO2 back into the tank, but then what would even use this event?

I don't mind adding events that are unused but might be useful eventually, but I'm not actually sure how this would ever be used.

Reply:
This'd be useful if you're currently inside one of those animal crates, which among other things have an internal atmosphere in '13.

Reply:
also mechs eventually